modules/react/common/lib/utils/changeFocus.ts

```diff
@@ -2,19 +2,23 @@
  * Programmatically change focus in a way that causes a focus ring around the new element.
  * @param element Element that should be focused
  */
-export const changeFocus = (element: HTMLElement | null) => {
+export const changeFocus = (element: unknown) => {
```

Comment:
It is possible to be given non-HTMLElement types. The use-case was a non-ref forwarding `MenuItem` component. So the ref was a `MenuItem` class instead of an `HTMLElement`. It is safer to just assume anything will be passed and deal with it in the function.

Reply:
I refreshed my understanding of the difference between `unknown` and `any`, but is there a reason to opt for it here? I don't have a preference, I'm just curious.

 You're not calling any actions on `element` until after checking that it's an instance of `HTMLElement`. Using `unknown` does ensure you don't get a false positive calling `focus`, and I suppose if someone modified this function later, it would be a little safer. Is that the reasoning here?

Reply:
`unknown` should almost always be preferred over `any`.

The main difference between `unknown` and `any`:

```tsx
const foo: any = null
foo.bar() // totally legal according to Typescript

const foo: unknown = null;
foo.bar(); // Object is of type 'unknown'. ts(2571)

if (typeof foo === 'object') {
  // object | null
  if (foo) {
    // object

    // define a guard
    function isFoo(input: unknown): input is {bar: Function} {
      return (
        typeof input === 'object' &&
        !!input &&
        input.hasOwnProperty('bar') &&
        typeof (input as any).bar === 'function'
      );
    }
    if (isFoo(foo)) {

      // { bar: Function }
      foo.bar();
    }
  }
}
```

`unknown` means the type is unknown and you must prove the type with runtime checks what the type is. `typeof` and `instanceof` are guard functions that determine the correct type. `unknown` is safer because of the runtime requirement to test what the type is. `any` will strip away any type checks.